### PR TITLE
Split stor into 4 modules, for packaging later

### PR DESCRIPTION
**Pull request by anujkumar93**
We want to split stor into 4 packages: stor, stor_swift, stor_dx, stor_s3. This is for ease of installation, so we don't need to install all dependencies if we want to use a single package. This PR initiates that process by separating out code for the four modules.

### CODE REVIEW

File: stor/obs.py [hunk 1]
```diff
@@ -457,10 +458,12 @@ def close(self):
             self._wait_on_close()
 
     def _wait_on_close(self):
```
[the remark this answers is not in the document]
Comment (jtratner) [reply stor/obs.py]: update: this is handled in upcoming PR. let's leave isinstance check in for now.

File: stor/obs.py [hunk 2]
```diff
@@ -457,10 +458,12 @@ def close(self):
...
-        if isinstance(self._path, stor.dx.DXPath):
+        # don't want to import stor_dx here because it is only installed as extra
+        # if isinstance(self._path, stor_dx.dx.DXPath):
```
Comment (jtratner) [inline stor/obs.py]: uncomment this line and then good to go

**Comment by jtratner** [comment]
in discussion, (3) isn't necessary because will be handled in follow up PR

File: stor/tests/test_utils.py [hunk 3]
```diff
@@ -9,10 +9,11 @@
 from testfixtures import LogCapture
 
 import stor
+import stor_swift
 from stor import Path
 from stor.posix import PosixPath
-from stor.s3 import S3Path
-from stor.swift import SwiftPath
+from stor_s3.s3 import S3Path
```
[the remark this answers is not in the document]
Comment (anujkumar93) [reply stor/tests/test_utils.py]: Agreed, looks cleaner. However in the interest of not changing code in this PR, and only moving files around, will do this in the other PRs I open

**Pull request merged**